### ISSUE TITLE
Cmder + Win10: ssh timeout

### ISSUE THREAD

**Issue by giulioprinaricotti** [deployphp/deployer#1920]
- I can easily ssh into the machine
 - Same project, with same script and configuration and keys works on my OSX laptop and Ubuntu machine
 - When I **dep deploy -vvv** from cmder on win10 this happens:

> λ php dep deploy -vvv

  [localhost] > git rev-parse --abbrev-ref HEAD
  [localhost] < master
  ✈︎ Deploying master on my.projecthome.com
  • done on [my.projecthome.com]
  ➤ Executing task deploy:prepare
  [my.projecthome.com] > echo $0
  [my.projecthome.com] < ssh multiplexing initialization
  [my.projecthome.com] < mux_client_request_session: read from master failed: Connection reset by peer
  [my.projecthome.com] < ControlSocket 
  /c/Users/gprin/.ssh/deployer_deployer@my.projecthome.com already exists, disabling multiplexing
  [my.projecthome.com] < bash
  [my.projecthome.com] > if [ ! -d /var/www/html/projecthome ]; then mkdir -p /var/www/html/projecthome; fi
  [my.projecthome.com] < mux_client_request_session: read from master failed: Connection reset by peer
  [my.projecthome.com] < ControlSocket /c/Users/gprin/.ssh/deployer_deployer@my.projecthome.com already exists, disabling multiplexing
  [my.projecthome.com] > if [ ! -L /var/www/html/projecthome/current ] && [ -d /var/www/html/projecthome/current ]; then echo 'true'; fi
  [my.projecthome.com] < mux_client_request_session: read from master failed: Connection reset by peer
  [my.projecthome.com] < ControlSocket /c/Users/gprin/.ssh/deployer_deployer@my.projecthome.com already exists, disabling multiplexing
[my.projecthome.com] > cd /var/www/html/projecthome && if [ ! -d .dep ]; then mkdir .dep; fi
[my.projecthome.com] < mux_client_request_session: read from master failed: Connection reset by peer
[my.projecthome.com] < ControlSocket /c/Users/gprin/.ssh/deployer_deployer@my.projecthome.com already exists, disabling multiplexing
[my.projecthome.com] > cd /var/www/html/projecthome && if [ ! -d releases ]; then mkdir releases; fi
[my.projecthome.com] < mux_client_request_session: read from master failed: Connection reset by peer

I've tried **disabling multiplexing** and this happens:

> λ php dep deploy -o ssh_multiplexing=false -vvv

[localhost] > git rev-parse --abbrev-ref HEAD
[localhost] < master
✈︎ Deploying master on my.projecthome.com
• done on [my.projecthome.com]
➤ Executing task deploy:prepare
[my.projecthome.com] > echo $0
[my.projecthome.com] < bash
[my.projecthome.com] > if [ ! -d /var/www/html/projecthome ]; then mkdir -p /var/www/html/projecthome; fi
[my.projecthome.com] > if [ ! -L /var/www/html/projecthome/current ] && [ -d /var/www/html/projecthome/current ]; then echo 'true'; fi
[my.projecthome.com] > cd /var/www/html/projecthome && if [ ! -d .dep ]; then mkdir .dep; fi
[my.projecthome.com] > cd /var/www/html/projecthome && if [ ! -d releases ]; then mkdir releases; fi
[my.projecthome.com] > cd /var/www/html/projecthome && if [ ! -d shared ]; then mkdir shared; fi
[my.projecthome.com] < ssh: connect to host my.projecthome.com port 22: Connection timed out
➤ Executing task deploy:failed
• done on [my.projecthome.com]
✔ Ok [1ms]
➤ Executing task deploy:unlock
[my.projecthome.com] > rm -f /var/www/html/projecthome/.dep/deploy.lock
[my.projecthome.com] < ssh: connect to host my.projecthome.com port 22: Connection timed out

I've already removed the OpenSSH client from Win10 as it wasn't able to connect to the machine somehow. Any hint? Love the project btw.

**Comment by that0n3guy** [comment]
I have this problem as well.   Deployer would get into ssh, create a couple folders on the server, then ssh would timeout while running.

I gave up trying to figure this out using the various windows bash options.   I tried many different configurations and bash options. 

To do this on windows 10, I just installed WSL (Windows subsystem for linux).... Here is a video on installing php on it (its just like ubuntu): https://www.jeffgeerling.com/blog/2018/installing-php-7-and-composer-on-windows-10-using-ubuntu-wsl (Here is microsofts docs: https://docs.microsoft.com/en-us/windows/wsl/install-win10).  You have access to the C: drive via /mnt/c/ in WSL and run deployer just fine.  

I pretty much just use WSL bash for deployer... Using gitbash for everything else because git in WSL is sloooow (https://github.com/microsoft/WSL/issues/873)
